### ISSUE TITLE
Feature request: Back up to a borg repo at an .onion address

### ISSUE THREAD

**Issue by andersaamodt** [borgbase/vorta#1275]
I tried to use Vorta for this previously and wasn't able to. It is possible to use borgbackup over tor with torify. It would be great if Vorta could do this, I would definitely use it then. Right now I my choice is between tor and a GUI which is a hard choice!

**Comment by real-yfprojects** [comment]
How do you torify borg? Do you have a guide for torification? And most importantly the torify HOWTO warns that one should know what one is doing when using torify. So how do you ensure torification works with borg and nothing is leak over DNS or such?

**Comment by andersaamodt** [comment]
The script I use has:

`export BORG_REPO="ssh://username@longonionaddress.onion:22/path/to/borg/repo/on/server/repo-name"
export BORG_RSH="torify ssh -i `/home/myusername/path/to/mysshkey"`

It also exports BORG_PASSPHRASE.

Then it just does a borg create command followed by a borg prune and outputs success as a system alert using notify-send.

Maybe this could be improved or updated by someone who knows tor better, but it works reliably and the traffic goes over tor so presumably it is pretty secure and better than no tor.

**Comment by andersaamodt** [comment]
Apparently it is best practice to use torsocks instead of torify, but they are identical. [documentation on torsocks](https://gitlab.torproject.org/legacy/trac/-/wikis/doc/torsocks)

**Comment by real-yfprojects** [comment]
> Apparently it is best practice to use torsocks instead of torify, but they are identical. [documentation on torsocks](https://gitlab.torproject.org/legacy/trac/-/wikis/doc/torsocks)

They are used identically but probably differ nonetheless.
You can set the `rsh` parameter by entering `--rsh "torify ssh -i /home/myusername/path/to/mysshkey"` in the LineEdit in schedule tab but this will only change the behaviour of `borg create`.

**Comment by real-yfprojects** [comment]
Maybe development on #379 can fix this issue too.

**Comment by andersaamodt** [comment]
Any movement on this?

**Comment by real-yfprojects** [comment]
Nope. If you have time, you could try to summarize #379 and identify further steps towards resolving the issue. I am thinking of tasks like making a feature list and description, answering open design questions and drawing a GUI mockup.

I think this issue can be resolved by resolving #379 since pre and post borg hooks could set environment variables like `BORG_RSH`.
